src/content/ui/adaptive-responsive/best-practices.md

```diff
@@ -0,0 +1,218 @@
+description: >
+  Learn the best practices for creating >
+  a responsive and an adaptive app.
+short-title: Best practices
+---
+
+The best practices on this page cover,
+in particular, large screen devices,
+but can equally apply to any Flutter app.
+Flutter defines [large screens][] as tablets,
+foldables, and ChromeOS devices running Android.
+
+:::secondary Why do large screens matter, in particular?
+Demand for large screens continues to increase.
+As of January 2024,
+more than [270 million active large screen][large screens]
+and foldable devices run on Android.
+When your app supports large screens,
+it also receives the following benefits:
+
+* Optimizing your app to fill the screen improves
+  your app's user engagement metrics.
+* Increases your app's visibility in the Play Store.
+  Recent [Play Store updates][] show ratings by
+  device type and indicates when an app lacks
+  large screen support. 
+:::
+
+[large screens]: {{site.android-dev}}/guide/topics/large-screens/get-started-with-large-screens
+[Play Store updates]: {{site.android-dev}}/2022/03/helping-users-discover-quality-apps-on.html
+
+These best practices cover responsive _and_
+adaptive app design.
+
+## SafeArea
+
+When running your app on the latest devices,
+you might encounter bits of the UI being blocked
+by cutouts on top of the device's screen.
+
+You can fix this with the [`SafeArea`][] widget,
+which insets its child widget to avoid intrusions
+(like notches and camera cutouts),
+as well as operating system UI (such as the status bar
+on Android, or by rounded corners of the display).
+The `SafeArea` widget allows you to enable or
+disable padding on any of its four sides.
+
+It’s generally recommended to wrap the body of a
+`Scaffold` widget in `SafeArea` as a safe place to start,
+but you don’t always need to put it this high in the
+`Widget` tree.
+
+<b>TODO(sfshaza): PENDING: example code</b>
+
+For example, if you purposefully want your app to stretch
+under the cutouts, you can move the `SafeArea` to wrap
+whatever content makes sense,
+and let the rest of the app take up the full screen.
+
+<b>TODO(sfshaza): PENDING: example</b>
+
+Using `SafeArea` ensures that your app content won’t be
+cut off by physical display features or operating system UI
+and sets your app up for success even as new devices with
+different shapes and styles of cutouts enter the market.
+
+[`SafeArea`]: {{site.api}}/flutter/widgets/SafeArea-class.html
+
+## Don't gobble up all horizontal space
+
+If viewing your mobile app on a large screen device,
+you might notice that either a box or some text might
+take up all the horizontal space.
+
+This is _not_ recommended by the Android Large Screen App
+Quality Guidelines.
+There are a couple ways you can solve this.
+
+You can use the `GridView` widget to transform your
+existing `ListView` into more reasonably sized items,
+or you can use the `maxWidth` property of `BoxConstraints`
+to not have the list go edge-to-edge.
+
```

Comment:
### Do not lock orientations

We strongly suggest you not lock screen orientations. Users do not like locked screen orientations. Locked orientation can be an [accessibility issue](https://www.w3.org/WAI/WCAG21/Understanding/orientation.html) for some users. Android large format tiers require portrait and landscape support at the [lowest level](https://developer.android.com/docs/quality-guidelines/large-screen-app-quality#T3-8). 
Android device manufacturers [can override](https://developer.android.com/guide/topics/large-screens/large-screen-compatibility-mode#per-app_overrides) locked orientation. 
Apple guidelines say ["Aim to support both portrait orientations" ](https://www.w3.org/WAI/WCAG21/Understanding/orientation.html).